src/event_queue.hpp

```diff
@@ -85,6 +103,22 @@ public :
         }
     }
 
+    // Pop and return top event `ev` of queue if `t_until` > `event_time(ev)`.
+    util::optional<value_type> pop_if_before(const time_type& t_until) {
+        using ::nest::mc::event_time;
+        return pop_if(
+            [&t_until](const value_type& ev) { return t_until > event_time(ev); }
+        );
+    }
+
+    // Pop and return top event `ev` of queue unless `event_time(ev)` > `t_until`
+    util::optional<value_type> pop_if_not_after(const time_type& t_until) {
```

Comment:
Here I think it is unavoidable; because we're limiting ourselves to the use of a strict inequality `>` on time types, we're making a distinction between 'before' (i.e. time is less than top of queue) and 'not after' (i.e. time is not greater than top of queue).

An alternative, `pop_if_before_or_same`, would imply we're testing for equality or greater-than-equal-to, when in truth we're still just using greater-than.